### DISCUSSION TITLE
What version of libLLVM should I develop against?

### DISCUSSION THREAD

**Discussion by Birch-san** [bpftrace/bpftrace#1782] [Development]
[Vagrantfile](https://github.com/iovisor/bpftrace/blob/master/Vagrantfile#L12) seems to compile & link against LLVM 8.

[`INSTALL.md`](https://github.com/iovisor/bpftrace/blob/master/INSTALL.md#building-bpftrace-1) seems to compile & link against LLVM 7.

I assume Vagrantfile is more recently-maintained?

Alternatively, can I develop against LLVM 12? Or do the fixture transformations in `test/codegen/common.h` only work on old LLVM?

**Comment by fbs** [comment]
Codegen tests are based on LLVM7 as they're somewhat annoying to update. Vagrant just uses whats available.

I'd recommend using LLVM12 for most things as it fixes #1305 . For the codegen tests you have to use LLVM7 for now which is a bit annoying but if you run docker its quite easy. Just use a dedicated build dir for it so you dont have to keep rebuilding.

**Reply by Birch-san** [reply]
is LLVM 8 fine too?

[`update_codegen_tests.sh`](https://github.com/iovisor/bpftrace/blob/master/scripts/update_codegen_tests.sh#L16) seems to use LLVM 8.
